R/g_mlm.R

```diff
@@ -54,18 +58,27 @@
 #' summary(Bryant2016_g1)
 #' print(Bryant2016_g1)
 #'
-#' Bryant2016_RML4 <- lme(fixed = outcome ~ session_c + treatment + trt_time,
-#'                        random = list(~ 1 | school, ~ session_c | case),
-#'                        correlation = corAR1(0, ~ session_c | school/case),
-#'                        data = Bryant2016)
-#' Bryant2016_g4 <- g_mlm(Bryant2016_RML4, p_const = c(0,0,1,5), r_const = c(1,1,0,0,0,1))
-#' summary(Bryant2016_g4)
-#' print(Bryant2016_g4)
+#' data(Laski, package = "scdhlm")
+#' Laski_AR1 <- gls(outcome ~ treatment,
+#'                  correlation = corAR1(0.2, ~ time | case),
+#'                  data = Laski)
+#' Laski_AR1_g <- g_mlm(Laski_AR1, p_const = c(0,1), r_const = c(0,1),
+#'                      infotype = "expected", returnModel = TRUE)
+#' summary(Laski_AR1_g)
+#' print(Laski_AR1_g)
 
 g_mlm <- function(mod, p_const, r_const, infotype = "expected", returnModel = TRUE) {
 
   # basic model estimates
-  p_beta <- sum(nlme::fixed.effects(mod) * p_const)               # p'Beta
+
+  if (class(mod) == "gls") {
```

Comment:
As a general practice, use `if (inherits(mod, "gls"))` instead of checking equality of class. For some models, `class(mod)` will be a vector with length > 1, which will cause `if (class(mod) == "something")` to throw an error.